src/Components/PostForm.jsx

```diff
@@ -0,0 +1,216 @@
+    let closeButton
+    if (this.state.successMessage === false) {
+      closeButton = (
+        <Button
+          id='close-topsidebar-error'
+          onClick={this.handleMessageVisibility('overlay')}
+        >
+          Close
+        </Button>
+      )
+    }
+    return (
+      <>
+        <Sidebar.Pushable as={Segment} textAlign='center'
+          className={this.state.activeItem}>
+          <TopSidebar
+            message={message}
+            visible={this.state.messageVisible}
+            successMessage={this.state.successMessage}
+            closeButton={closeButton}
+          />
+          <Sidebar.Pusher dimmed={this.state.messageVisible}>
+            <Container id="upload-post-wrapper">
+              <Header as='h3'>Add a photo</Header>
+              <ImageUploader
+                buttonText={
+                  <div>
+                    <p id="add-photo-headline">Add Image</p>
+                    <Icon id="add-photo-icon" name="image outline" size="huge"></Icon>
+                    <p id="add-photo-label">Maximum image file size: 5 MB, Accepted image types: JPG</p>
+                  </div>
+                }
+                buttonClassName={this.state.button}
+                withLabel={false}
+                withIcon={false}
+                withPreview={true}
+                singleImage={true}
+                onChange={this.onImageDropHandler}
+                imgExtension={['.jpg']}
+                maxFileSize={5242880}
+              />
+              <Form size="mini" type='medium'>
+                <Form.Input
+                  required
+                  id="caption"
+                  value={this.state.caption}
+                  onChange={this.onChangeHandler}
+                  placeholder="Write your caption here"
+                />
+              </Form>
+              <p id="location">
+                <Icon
+                  name='map marker alternate' />
+                Södermalm, Swedenborgsgatan</p>
+              <br></br>
+              <Button.Group
+                toggle={true}
+                inverted={true}>
+                <Button
+                  id='work'
+                  basic color='teal'
+                  active={activeItem === 'work'}
+                  value='work'
+                  onClick={this.handleChangeCategory}>
+                  WORK
+          </Button>
+                <Button
+                  id='play'
+                  basic color='yellow'
+                  active={activeItem === 'play'}
+                  value='play'
+                  onClick={this.handleChangeCategory}>
+                  PLAY
+            </Button>
+              </Button.Group>
+              <br></br>
+              <br></br>
+              <Button id="upload-button" onClick={this.uploadPost}>MAP IT!</Button>
+            </Container>
+          </Sidebar.Pusher>
+        </Sidebar.Pushable>
+      </>
+    )
+  }
+}
+export default PostForm
```

Comment:
Please revisit this file. I have a hard time reviewing it. Way too long...

Reply:
Yes, see comment above!